inc/inventory/request.class.php

```diff
@@ -346,14 +416,39 @@ public function getResponse() :string {
          throw new \RuntimeException("Mode has not been set");
       }
 
+      $data = null;
       switch ($this->mode) {
          case self::XML_MODE:
-            return $this->response->saveXML();
+            $data = $this->response->saveXML();
+            break;
          case self::JSON_MODE:
-            return json_encode($this->response);
+            $data = json_encode($this->response);
+            break;
          default:
             throw new \RuntimeException("Unknown mode " . $this->mode);
+            break;
       }
+
+      if ($this->compression !== self::COMPRESS_NONE) {
+         switch ($this->compression) {
+            case self::COMPRESS_ZLIB:
+               $data = gzcompress($data);
+               break;
+            case self::COMPRESS_GZIP:
+                $data = gzencode($data);
+                break;
+            case self::COMPRESS_BR:
+               if (!function_exists('brotli_compress')) {
+                   throw new \UnexpectedValueException("You must install Brotli PHP extension.");
+               }
+               $data = brotli_compress($data);
```

Comment:
Is it possible here to also return an error to the agent telling this compression is not supported as "content-type" ? Maybe a simple "400 brotli not supported" or "[415 Unsupported Media Type](https://developer.mozilla.org/fr/docs/Web/HTTP/Status/415)". So the agent can fall back on zlib or gzip content type ?
Anyway brotli is still not supported on agent side at the moment.

Reply:
I've done that, but when setting compression; and I've remove the exception here, since it should not happen.